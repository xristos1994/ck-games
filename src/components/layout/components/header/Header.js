import React from "react";
import { Link } from "gatsby";
import { GiHamburgerMenu } from "react-icons/gi";
import { classnames } from "@utils/component-utils";
import { useScrollDirection } from "@hooks";
import { Button } from "@components";
import styles from "./styles.module.css";

const HeightClasses = {
  normal: "normal",
  shrinked: "shrinked",
};

const Header = () => {
  const scrollDirection = useScrollDirection({
    initialDirection: useScrollDirection.ScrollDirections.up,
    thresholdPixels: 40,
  });

  const heightClass =
    scrollDirection === useScrollDirection.ScrollDirections.up
      ? HeightClasses.normal
      : HeightClasses.shrinked;

  return (
    <header className={classnames(styles.header, styles[heightClass])}>
      <Link to="/" className={styles.title}>
        My Games
      </Link>
      <Button className={styles.menuButton}>
        <GiHamburgerMenu />
      </Button>
    </header>
  );
};

export { Header };
