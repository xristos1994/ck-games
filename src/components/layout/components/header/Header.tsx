import React, { FC, ReactElement } from "react";
import { Link } from "gatsby";
import Logo from "@images/svg/flatLogo.svg";

const styles = require("./styles.module.css");

const Header: FC = (): ReactElement => {
  return (
    <header className={styles.header}>
      <Link to="/" className={styles.logoContainer}>
        <Logo className={styles.logo} />
      </Link>
    </header>
  );
};

export { Header };
