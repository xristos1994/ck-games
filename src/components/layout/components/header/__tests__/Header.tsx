import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from './../Header';

const FlatLogoIcon = () => {
  return (
    <div>
      FlatLogoIcon
    </div>
  );
};

jest.mock('./../../../../icons', () => ({
  __esModule: true,
  FlatLogoIcon: FlatLogoIcon
}));

describe('components/Header', () => {
  it('renders correctly', () => {
    global.window = {
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      location: {
        pathname: ''
      }
    };

    const tree = renderer.create(
      <Header />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
