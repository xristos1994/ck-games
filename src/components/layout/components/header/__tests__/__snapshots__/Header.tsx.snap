// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header renders correctly 1`] = `
<header
  className="header"
>
  <a
    aria-current="page"
    className="logoContainer"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    style={Object {}}
  >
    <div>
      FlatLogoIcon
    </div>
  </a>
</header>
`;
